src/poliastro/ephem.py

```diff
@@ -50,46 +49,42 @@ def build_ephem_interpolant(body, period, t_span, rtol=1e-5):
 
     Returns
     -------
-    intrp : ~scipy.interpolate.interpolate.interp1d
-        Interpolated function.
+    interpolant : callable
+        Interpolant function that receives time increment in seconds
+        since the initial epoch.
 
     """
-    h = (period * rtol).to_value(u.d)
-    t_span = (t_span[0].to_value(u.d), t_span[1].to_value(u.d) + 0.01)
-    t_values = np.linspace(*t_span, int((t_span[1] - t_span[0]) / h))  # type: ignore
-    r_values = np.zeros((t_values.shape[0], 3))
-
-    for i, t in enumerate(t_values):
-        epoch = Time(t, format="jd", scale="tdb")
-
-        r = get_body_barycentric(body.name, epoch)
-        r = (
-            ICRS(x=r.x, y=r.y, z=r.z, representation_type=CartesianRepresentation)
-            .transform_to(GCRS(obstime=epoch))
-            .represent_as(CartesianRepresentation)
-        )
+    epochs = time_range(
+        Time(t_span[0], format="jd", scale="tdb"),
+        end=Time(t_span[1], format="jd", scale="tdb"),
+        periods=int(((t_span[1] - t_span[0]) / (period * rtol)).to_value(u.one)),
+    )
+    ephem = Ephem.from_body(body, epochs, attractor=attractor)
 
-        r_values[i] = r.xyz.to(u.km)
+    def interpolant(t):
```

Comment:
This probably poses a big performance regression and can't be merged as it stands now 😭